streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -647,52 +626,47 @@ private void addRecordsToTasks(final ConsumerRecords<byte[], byte[]> records) {
     /**
      * Schedule the records processing by selecting which record is processed next. Commits may
      * happen as records are processed.
-     * @param tasks The tasks that have records.
      * @param recordsProcessedBeforeCommit number of records to be processed before commit is called.
      *                                     if UNLIMITED_RECORDS, then commit is never called
      * @return Number of records processed since last commit.
      */
-    private long processAndPunctuate(final Map<TaskId, StreamTask> tasks,
-                                     final long recordsProcessedBeforeCommit) {
+    private long processAndPunctuate(final long recordsProcessedBeforeCommit) {
 
-        long totalProcessedEachRound;
-        long totalProcessedSinceLastMaybeCommit = 0;
+        final AtomicLong totalProcessedEachRound = new AtomicLong(0);
```

Comment:
because they need to be final so they can be accessed by the anonymous inner class